bottles/backend/managers/manager.py

```diff
@@ -410,35 +410,35 @@ def check_runners(self, install_latest: bool = True) -> bool:
         self.runners_available = sorted(self.runners_available, reverse=True)
         return True
 
+    @RunAsync.run_async
     def check_runtimes(self, install_latest: bool = True) -> bool:
-        self.runtimes_available = []
+        RepoStatus.repo_wait_operation("components.fetching")
+        RepoStatus.repo_wait_operation("components.organizing")
         if "FLATPAK_ID" in os.environ:
-            self.runtimes_available = ["flatpak-managed"]
+            self.available_runtime = "flatpak-managed"
             return True
 
         runtimes = os.listdir(Paths.runtimes)
-
+        latest_version = next(iter(self.supported_runtimes))
         if len(runtimes) == 0:
-            if install_latest and self.utils_conn.check_connection():
-                logging.warning("No runtime found.")
-                try:
-                    version = next(iter(self.supported_runtimes))
-                    return self.component_manager.install("runtime", version)
-                except StopIteration:
-                    return False
-            return False
-
-        runtime = runtimes[0]  # runtimes cannot be more than one
-        manifest = os.path.join(Paths.runtimes, runtime, "manifest.yml")
+            logging.warning("No runtime found.")
+            install_latest = True
+        else:
+            runtime = runtimes[0]  # runtimes cannot be more than one
```

Comment:
Are you sure? If users have 0.5 installed, then 0.6 will be installed. But then, both will be in the folder? There is nothing that deletes old version, unless I'm mistaken?